analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+
+public class Program
+{
+    public void Base(string[] myArray)
+    {
+        Method(new string[] { "s1", "s2" }); // Noncompliant {{Remove this array creation and simply pass the elements.}}
+//             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        Method(new string[] { "s1" }); // Noncompliant
+        Method("s1");           // Compliant
+        Method("s1", "s2");     // Compliant
+        Method(myArray);        // Compliant
+        Method(new string[12]); // Compliant
+
+        Method2(1, new string[] { "s1", "s2" }); // Noncompliant {{Remove this array creation and simply pass the elements.}}
+//                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        Method2(1, new string[] { "s1" }); // Noncompliant
+        Method2(1, "s1");           // Compliant
+        Method2(1, "s1", "s2");     // Compliant
+        Method2(1, myArray);        // Compliant
+        Method2(1, new string[12]); // Compliant
+
+        Method3(new string[] { "s1", "s2" }, "s1"); // Compliant
+        Method3(new string[] { "s1", "s2" }, new string[12]); // Compliant
+        Method3(new string[] { "s1", "s2" }, new string[] { "s1", "s2" }); // Noncompliant
+//                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        Method3(null, null);        // Compliant
+        Method4(new [] { "s1" });   // Compliant
+        Method4(new [] { "s1", "s2" }); // Compliant
+
+        Method3(args: new string[] { "s1", "s2" }, a: new string[12]); // Compliant (if you specifically require the arguments to be passed in this order there is no way of making this compliant, thus we shouldn't raise)
+        Method3(args: new string[12], a: new string[] { "s1", "s2" }); // Compliant
+
+        var s = new MyClass(1, new int[] { 2, 3 }); // Noncompliant
+//                             ^^^^^^^^^^^^^^^^^^
+        var s1 = new MyClass(1, 2, 3); // Compliant
```

Comment:
I just realized: Target typed new is missing. Unfortunately, this requires a new test file, as it is a C#9 feature.

```cs
        MyClass s = new(1, new int[] { 2, 3 }); // Noncompliant
```